usautobuild/logger.py

```diff
@@ -1,39 +1,127 @@
+        "WARNING": logging.WARNING,
+        "ERROR": logging.ERROR,
+    }
+
+    return level_name_map.get(s.upper(), logging.INFO)
+
 
 def setup_logger(arg_level: str):
+    """Configure basic logging facilities"""
+
+    log.setLevel(str_to_log_level(arg_level))
+
+    log_path = Path("logs")
+    log_path.mkdir(exist_ok=True)
+
+    fmt = logging.Formatter("[%(asctime)s::%(name)s::%(levelname)s] %(message)s")
+
+    for handler in (
+        logging.StreamHandler(sys.stdout),
+        handlers.RotatingFileHandler(
+            log_path / datetime.datetime.now().strftime("%y-%m-%d-%H.log"),
+            maxBytes=(1048576 * 5),
+            backupCount=7,
+        ),
+    ):
+        handler.setFormatter(fmt)
+        log.addHandler(handler)
 
-    log = logging.getLogger('usautobuild')
-    log.setLevel(get_log_level(arg_level))
-
-    format = logging.Formatter("[%(asctime)s::%(name)s::%(levelname)s] %(message)s")
-
-    ch = logging.StreamHandler(sys.stdout)
-    ch.setFormatter(format)
-    log.addHandler(ch)
-
-    fh = handlers.RotatingFileHandler(
-        f"logs/{datetime.datetime.now().strftime('%y-%m-%d-%H.log')}",
-        maxBytes=(1048576 * 5),
-        backupCount=7)
-    fh.setFormatter(format)
-    log.addHandler(fh)
-
-def get_log_level(arg_level: str):
-    if arg_level:
-        if arg_level.upper() == "DEBUG":
-            log_level = logging.DEBUG
-        elif arg_level.upper() == "INFO":
-            log_level = logging.INFO
-        elif arg_level.upper() == "WARNING":
-            log_level = logging.WARNING
-        elif arg_level.upper() == "ERROR":
-            log_level = logging.ERROR
+
+def setup_extra_loggers(config: Config):
+    """Configure complex loggers requiring config"""
+
+    discord_webhook = config.discord_webhook
+    if discord_webhook is not None:
+        handler = DiscordHandler(discord_webhook)
+        handler.setLevel(logging.INFO)
+        log.addHandler(handler)
+
+
+class DiscordHandler(logging.Handler):
+    uwu_replacements = {
+        "r": "w",
+        "!": "! owo",
+        "v": "w",
+        "ni": "nye",
+        "na": "nya",
+        "ne": "nye",
+        "no": "nyo",
+        "nu": "nyu",
+        "ove": "uv",
+        "l": "w",
+    }
+    uwu_prefixes = (
+        "Nya~",
+        "Nnnnya~",
+    )
+    uwu_endings = (
+        "OwO",
+        "UwU",
+        "~",
+        "~~",
+    )
+
+    def __init__(self, url: str):
+        super().__init__()
+
+        self.url = url
+
+    @classmethod
+    def uwuize(cls, message: str) -> str:
+        for key, value in cls.uwu_replacements.items():
+            message = re.sub(key, value, message, flags=re.IGNORECASE)
+
+        if random.random() < .8:
+            message = f"{random.choice(cls.uwu_prefixes)} {message}"
+
+        if random.random() < .8:
+            message = f"{message} {random.choice(cls.uwu_endings)}"
+
+        return message
+
+    def send_message(self, message: str, fail: bool = False) -> bool:
```

Comment:
Aw I was hoping you improved this method. I never liked passing the fail state as a bool.

Reply:
not sure what you are talking about. better name for parameter? i don't have any ideas now. there is less code now so it should be easier to change it later